gems/sorbet-runtime/lib/types/types/typed_enumerable.rb

```diff
@@ -139,7 +139,8 @@ def describe_obj(obj)
         # enumerating the object is a destructive operation and might hang.
         obj.class
       else
-        self.class.new(type_from_instances(obj))
+        # Get the real class name for other `Enumerable` as we don't want them to be collapsed as a `T::Array[T.untyped]`
+        obj.class
```

Comment:
this fails typechecking.
```suggestion
        T.unsafe(obj).class
```